app/web/src/store/func/funcs.store.ts

```diff
@@ -648,13 +593,52 @@ export const useFuncStore = () => {
             callback: (data) => {
               if (data.changeSetId !== selectedChangeSetId) return;
               this.FETCH_FUNC_LIST();
+
+              const assetId = assetStore.getLastSelectedAssetId();
+              if (
+                assetId &&
+                assetStore.selectedFuncId &&
+                assetStore.selectedFuncId === this.selectedFuncId
+              ) {
+                assetStore.closeFunc(assetId, this.selectedFuncId);
+              }
             },
           },
           {
             eventType: "FuncSaved",
```

Comment:
The response body based logic for both routes involving saving funcs now resides in `FuncSaved`